core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -133,7 +149,10 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
   private def endpoints = config.listeners.map(l => l.listenerName -> l).toMap
 
   private def createAcceptorAndProcessors(processorsPerListener: Int,
-                                          endpoints: Seq[EndPoint]): Unit = synchronized {
+    endpoints: Seq[EndPoint],
+    requestChannel: RequestChannel,
+    processorCollector: java.util.Map[Int, Processor],
+    isControlPlane: Boolean): Unit = synchronized {
```

Comment:
This is essentially only used for naming the KafkaThread. What do you think about doing this : 
```suggestion
  private def createDataPlaneAcceptorsAndProcessors(processorsPerListener: Int,
                                                    endpoints: Seq[EndPoint]) : Unit = synchronized {
    endpoints.foreach { endpoint =>
      val acceptor = createAcceptorWithProcessors()
      addProcessors(acceptor, endpoint, processorsPerListener, dataRequestChannel, dataProcessors)
      KafkaThread.nonDaemon(s"data-plane-kafka-socket-acceptor-${endpoint.listenerName}-${endpoint.securityProtocol}-${endpoint.port}", acceptor).start()
      acceptor.awaitStartup()
      acceptors.put(endpoint, acceptor)
    }
  }

  private def createControlPlaneAcceptorAndProcessor () : Unit = synchronized {
    if (config.controlPlaneListener.isDefined) {
      val controlPlaneEndPoint = config.controlPlaneListener.get
      controlPlaneRequestChannelOpt = Some(new RequestChannel(20, RequestChannel.ControlPlaneRequestQueueSizeMetric))
      val controlPlaneAcceptor = createAcceptor(controlPlaneEndPoint)
      val controlPlaneProcessor = newProcessor(nextProcessorId, controlPlaneRequestChannelOpt.get, connectionQuotas, controlPlaneEndPoint.listenerName, controlPlaneEndPoint.securityProtocol, memoryPool)
      controlPlaneRequestChannelOpt.get.addProcessor(controlPlaneProcessor)
      val listenerProcessors = new ArrayBuffer[Processor]()
      listenerProcessors += controlPlaneProceesor
      controlPlaneAcceptor.addProcessors(listenerProcessors)
      KafkaThread.nonDaemon(s"control-plane-kafka-socket-acceptor-$controlPlaneListenerName-$controlPlaneSecurityProtocol-${controlPlaneEndPoint.port}", controlPlaneAcceptor).start()
      controlPlaneAcceptor.awaitStartup()
      acceptors.put(controlPlaneEndPoint, controlPlaneAcceptor)
    }
  }

  private def createAcceptor(endpoint: EndPoint): Acceptor = synchronized {
    val sendBufferSize = config.socketSendBufferBytes
    val recvBufferSize = config.socketReceiveBufferBytes
    val brokerId = config.brokerId
    new Acceptor(endpoint, sendBufferSize, recvBufferSize, brokerId, connectionQuotas)
  }
```